common/models/entities_with_tags.go

```diff
@@ -0,0 +1,43 @@
+package models
+
+type EntitiesWithTags []*EntityWithTags
+
+type ewtFilter func(e EntityWithTags) bool
+
+func (e EntitiesWithTags) RemoveIf(f ewtFilter) EntitiesWithTags {
```

Comment:
If this was a `models` package that was intended to be used by other language consumers, then I think being hardline would make more sense. But it isn't, and the benefits of the current semantics for `getNameTag` outweigh rigidity (i think)